File: Python/longest_substring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        x = [*s]
        if len(x) == 0:
            return 0
        max_length = 0
        
        for i in range(len(x)):
            letter_stack= [x[i]]
            temp_count = 1
            j = i+1
            while j < len(x):
                letter_stack.append(x[j])
                if len(set(letter_stack)) == len(letter_stack):
                    temp_count += 1
                    j += 1
                else:
                    break
            max_length = max(max_length, temp_count)
            # print(letter_stack)
            
        return max_length

File: Cpp/longest_substring.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int j=0, longest_len = 0;
        unordered_set<char> seen_str;

        for (int i=0; i<s.size(); i++){
            while (seen_str.find(s[i]) != seen_str.end()){
                seen_str.erase(s[j]);
                j++;
            }
            longest_len = max(longest_len, i - j + 1);
            seen_str.insert(s[i]);
        }
        return longest_len;
        
    }
};